src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,418 @@
+
+        /**
+         * Builder for the {@link CollectdRecord} class.
+         */
+        public static final class Builder extends OvalBuilder<CollectdRecord> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(CollectdRecord::new);
+            }
+
+            /**
+             * Sets the host.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setHost(final String value) {
+                _host = value;
+                return this;
+            }
+
+            /**
+             * Sets the time.  Time value is floating point epoch seconds.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setTime(final double value) {
+                _time = value;
+                return this;
+            }
+
+            /**
+             * Sets the plugin.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setPlugin(final String value) {
+                _plugin = value;
+                return this;
+            }
+
+            /**
+             * Sets the plugin instance.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            @JsonProperty("plugin_instance")
+            public Builder setPluginInstance(final String value) {
+                _pluginInstance = value;
+                return this;
+            }
+
+            /**
+             * Sets the type.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setType(final String value) {
+                _type = value;
+                return this;
+            }
+
+            /**
+             * Sets the sample values.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setValues(final List<Double> value) {
+                _values = value;
+                return this;
+            }
+
+            /**
+             * Sets the sample DS types.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            @JsonProperty("dstypes")
+            public Builder setDsTypes(final List<String> value) {
+                _dsTypes = value;
+                return this;
+            }
+
+            /**
+             * Sets the sample DS names.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            @JsonProperty("dsnames")
+            public Builder setDsNames(final List<String> value) {
+                _dsNames = value;
+                return this;
+            }
+
+            /**
+             * Sets the type instance.
```

Comment:
Can you add optional/required and any constraints and/or defaults to the setter comments?